yew-functional/src/lib.rs

```diff
@@ -1,3 +1,66 @@
+//! Function components are a simplified version of normal components.
```

Comment:
I haven't really gone through the docstrings yet because this review is already long enough as is.
It looks like you recycled a lot of content from the website though, which isn't ideal because it makes it much harder to maintain.
The module level documentation here, for example, could be replaced with a link to the Yew docs (because it's essentially just a slimmed down version of that).

I'm not sure what the best approach here is, but maybe we should keep the docstrings down to like one or two sentences, an example, and maybe a link to the website?
Let me know if you have an idea.

Reply:
>  The module level documentation here, for example, could be replaced with a link to the Yew docs (because it's essentially just a slimmed down version of that).

That is true. It might be better to link to website but that's not something we can do because currently, the links to functional components will 404 on `yew.rs`. The same will be true if the documentation link are changed in the future.

Do you think it's fine to link the `next` version right now? If it is, it might be a better idea to keep the doc strings to a sentence or two describing the Hook, link to website and example which is to be tested.

Reply:
The maintenance cost is too much for having docstrings and docs on website being too similar. I'll just slim down the docstrings (but keep the examples as-is so they can be tested)

Reply:
Yeah that's much better. Maybe we can do something about the links in the future too (so that it automatically links to the website reflecting the current version).